front-end/src/maple-parser/MapleGrammar.g4

```diff
@@ -86,8 +86,9 @@ functionDefinition : (TYPE | TYPE_VOID) ID '(' (typeList | TYPE_VOID)? ')' block
 returnStatement : 'return' expr SC ;
 blockFunction : '{' declaration* instruction* '}' ;
 argumentList : ((expr ',')* expr)? ;
-typeList : ((TYPE ID ',')* TYPE ID) ;
-
+typeList : ((argumentTypeVar | argumentTypeArray) ',')*  (argumentTypeVar | argumentTypeArray);
+argumentTypeVar : TYPE ID;
```

Comment:
Add `argumentType: argumentTypeVar | argumentTypeArray ;`

visitTypeList: `for (item : ctx->argumentType()) list.push(visit(item));`

visitArgumentType: `if (ctx->argumentTypeVar) return (FunctionParam*) visit(ctx->..; return (FunctionParam*) visit(ctx->argumentTypeArray)`

Reply:
Done